src/reanimated2/animation/repeat.ts

```diff
@@ -100,7 +105,13 @@ export const withRepeat = function <T extends AnimationObject>(
         now: Timestamp,
         previousAnimation: Animation<any> | null
       ): void {
-        animation.startValue = value;
+        // Detect re-render
+        const hasBeenInterrupted = value !== initialAnimationStartValue;
+        const startingValue =
+          hasBeenInterrupted && initialAnimationStartValue !== undefined
+            ? initialAnimationStartValue
+            : value;
+        animation.startValue = startingValue;
```

Comment:
```suggestion
        const wasInterrupted = value !== initialAnimationStartValue;
        const startingValue =
          wasInterrupted && initialAnimationStartValue !== undefined
            ? initialAnimationStartValue
            : value;
        animation.startValue = startingValue;
```

Also, why do we check for `undefined` here? We don't do it in case of `withSequence` and `withDelay`.